homeassistant/components/bluetooth/update_coordinator.py

```diff
@@ -88,23 +90,13 @@ def available(self) -> bool:
         """Return if the device is available."""
         return self._available
 
-    @callback
-    def _async_handle_bluetooth_event_internal(
-        self,
-        service_info: BluetoothServiceInfoBleak,
-        change: BluetoothChange,
-    ) -> None:
-        """Handle a bluetooth event."""
-        self._available = True
-        self._async_handle_bluetooth_event(service_info, change)
-
```

Comment:
This is now handled in the two subclasses to ensure it knows what the previous value was before it flips it